addons/osfstorage/models.py

```diff
@@ -160,16 +160,29 @@ def delete(self, user=None, parent=None, **kwargs):
         self._materialized_path = self.materialized_path
         return super(OsfStorageFileNode, self).delete(user=user, parent=parent) if self._check_delete_allowed() else None
 
+    def update_folder_regions(self, destination_parent):
+        for child in self.children.all():
+            if child.is_file:
+                child.update_most_recent_fileversion(destination_parent)
+            else:
+                child.update_folder_regions(destination_parent)
+
+    def update_most_recent_fileversion(self, destination_parent):
```

Comment:
Same with this method: if this method only makes sense on files (not folders), then it should be moved to `OsfStorageFile`.

Reply:
Actually, it might make sense to implement the same method for both folders and files, and then call that method in `move_under`.

```python
def update_region_from_latest_version(self, destination_parent):
    raise NotImplementedError

def move_under(...):
    # ...
    # no need to check is_file any more
    self.update_region_from_latest_version(destination_parent)
```

(this is the [Composite design pattern](https://www.wikiwand.com/en/Composite_pattern))